scout/server/blueprints/institutes/controllers.py

```diff
@@ -424,13 +424,13 @@ def populate_case_obj(case_obj):
             store.user(user_email) for user_email in case_obj.get("assignees", [])
         ]
         # Check if case was re-runned
-        analyses = case_obj.get("analyses", [])
-        now = datetime.datetime.now()
-        case_obj["is_rerun"] = (
-            len(analyses) > 1
-            or analyses
-            and analyses[0].get("date", now) < case_obj.get("analysis_date", now)
-        )
+        last_analysis = case_obj.get("analysis_date", datetime.datetime.now())  # it's a datetime
+        all_analyses = set()  # a set of datetimes
+        for analysis in case_obj.get("analyses", [{"date": last_analysis}]):
+            all_analyses.add(analysis.get("date", last_analysis))
+
+        case_obj["is_rerun"] = len(all_analyses) > 1 or last_analysis > max(all_analyses)
```

Comment:
Why max if len == 0 or 1?

Reply:
It was written like that in the code before (kind of). But I think it is sufficient to just have this:
`if last_analysis_date > max(all_analyses_dates)`

I'll test with this code on stage!

Reply:
The code had a bug before, and I remember that in some cases we could have the last analysis (case["analysis_date]) also inside the analyses list.  I think that's why we have 1 there

Reply:
Sorry, I don't follow? Either its longer than 1, or we compare to the max value of the one (or 0)?

Reply:
With the bug from some time ago (current analysis in analyses list) last_analysis > the_only_analysis might be False

Reply:
In that case you need len(all_analyses) > 1

Reply:
But its an `or` not an `xor`! If one of them is true, the expression will be true. So if there is more than one analysis, it will be true regardless of what is in the second term. Hence enough to test the one remaining value.

Reply:
What is `xor`?

What if you have the following situation saved in database (old bug):

```
case.analysis_date : 2022-10-04
case.analyses : [{date:2021-10-04}, {date:2022-10-04}] 
```

Then if you check with ONLY 
if `last_case_analysis > all_cases_analyses`

You get False. 
But it's rerun, because it has 2 analyses

Or am I not understanding your question?

Reply:
No, I think not? The issue is the use of the `max()` function. First you test if there are more than one unique date. Then you take the max of all the unique dates. Thing is, if there is more than one, the whole expression is true anyway!

Reply:
To sum up the discussion: we leave it as it is since it works. When we touch the code again, modify this line to:

case_obj["is_rerun"] = len(all_analyses) > 1 or last_analysis > all_analyses[0]